cmd/deploy/start.go

```diff
@@ -215,19 +215,28 @@ func WithLocalFile(cmd *cobra.Command, options *deployStartOptions, deployClient
 	if present && !isATest {
 		options.deploymentFile = gitWorkspace + options.deploymentFile
 	}
+
 	// read yaml file
 	file, err := os.ReadFile(options.deploymentFile)
 	if err != nil {
 		return nil, nil, errorUtils.NewWrappedError(ErrYAMLFileRead, err)
 	}
-	validationFailures := validate.Validate(file)
-	validate.LogValidationErrors(cmd.OutOrStdout(), validationFailures, false)
-	cmd.SilenceUsage = true
+
 	// unmarshall data into struct
 	var payload map[string]any
 	if err = yaml.Unmarshal(file, &payload); err != nil {
 		return nil, nil, errorUtils.NewWrappedError(ErrInvalidDeploymentObject, err)
 	}
+
+	// TODO: because the cli's cue schema is static it doesn't validate lambda functions. For the moment lets not validate if the kind is lambda.
+	if payload["kind"] != "lambda" {
```

Comment:
The magic strings don't spark joy in me 😄 

I think this check probably belongs in the `validate.Validate(...)` method. Why: ask yourself what code will end up being changed when you delete this TODO - it's definitely going to be something in `validate.Validate`. Ideally you won't need to change anything in this file.

Reply:
Inside `validate.Validate`:

```
var requestKind struct { Kind `json:"kind"` }
if err := json.Unmarshal(fileBytes, &requestKind); err != nil {
   // blah blah blah
}

if requestKind.Kind == "kubernetes" {
  // do validation
}
```

Reply:
Yes, good catch. Implemented using your suggestion.